lib/hyacinth/digital_object/type_def/dynamic_field_data.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module Hyacinth
+  module DigitalObject
+    module TypeDef
+      class DynamicFieldData < Hyacinth::DigitalObject::TypeDef::JsonSerializableCollection
+        TYPE_TO_FORM_TYPE = {
+          rights_metadata: ['item_rights', 'asset_rights'],
+          descriptive_metadata: ['descriptive']
+        }.freeze
+
+        def initialize(type)
+          super()
+          raise ArgumentError, "type '#{type}' is invalid" unless TYPE_TO_FORM_TYPE.keys.map(&:to_s).include?(type.to_s)
+
+          @type = type.to_sym
+        end
+
+        def from_serialized_form_impl(json_var)
+          return nil if json_var.nil?
+          raise ArgumentError, "Expected hash, but got: #{json_var.class}" unless json_var.is_a?(Hash)
+
+          terms_map = extract_terms(field_map, json_var)
+
+          search_query = terms_map.map { |vocab, terms|
```

Comment:
Maybe it would be worth moving the batch search functionality into the term search adapter itself (in a new method like `batch_search` that takes an argument that's a hash like {names: ['uri1', 'uri2'], collections: ['uri3', 'uri4']), since it'll be useful to look up a bunch of terms in a vocabulary for both validation purposes AND, later, new term registration or existing-term-enhancement purposes.

Reply:
I'll look into a refactor. I need think more about how creating and update terms is going to happen.

Reply:
Great, thanks.

Reply:
I would rather do this refactor as part of the create and update terms addition. I think there's other code that needs to be extracted out in addition to the method you pointed out.

Reply:
Ticket sounds good!